contracts/Market.sol

```diff
@@ -389,26 +385,52 @@ contract Market is IMarket, Ownable, ERC721 {
 		return _bets[index].payout;
 	}
 
-
 	function _payout(uint256 index, uint8 result) internal virtual {
-		if (result == SCRATCHED) {
-			uint256 lay = _bets[index].payout - _bets[index].amount;
+		uint256 payout = _bets[index].payout;
+		uint256 amount = _bets[index].amount;
+		uint256 loan = payout - amount;
+
+		// Deduct from total exposure first
+		_totalExposure -= loan;
+
+		if (result == SCRATCHED) {				
 			// Transfer the bet amount to the owner of the NFT
-			IERC20(_vault.asset()).transfer(ownerOf(index), _bets[index].amount);
-			// Transfer the lay back to the vault
-			IERC20(_vault.asset()).transfer(address(_vault), lay);
-		} else {
-			address recipient = result == WINNER ? ownerOf(index) : address(_vault);
-			uint256 amount = _bets[index].payout;
+			IERC20(_underlying).transfer(ownerOf(index), amount);
+			
+			// Transfer the loaned amount back to the vault
+			IERC20(_underlying).transfer(address(_vault), loan);
+			emit Repaid(address(_vault), loan);
```

Comment:
shouldnt these events come before the external interactions?

Reply:
If the transfer fails, you don't really want to emit the event though @brettvniekerk

Reply:
Nah, should be after the transfer so we know its been done.